protocol/flows/src/v5/ibd/flow.rs

```diff
@@ -132,7 +132,20 @@ impl IbdFlow {
         self.sync_missing_block_bodies(&session, relay_block.hash()).await?;
 
         // Following IBD we revalidate orphans since many of them might have been processed during the IBD
-        self.ctx.revalidate_orphans(&session).await;
+        // or are now processable
+        let (hashes, tasks) = self.ctx.revalidate_orphans(&session).await;
+        let mut unorphaned_hashes = Vec::with_capacity(hashes.len());
+        let results = join_all(tasks).await;
+        for (hash, result) in hashes.into_iter().zip(results) {
+            match result {
+                Ok(_) => unorphaned_hashes.push(hash),
+                Err(e) => warn!("Validation failed for orphan block {}: {}", hash, e),
+            }
+        }
+        match unorphaned_hashes.len() {
```

Comment:
Isn't it simpler to write `if unorphaned_hashes.len() != 0`?

Reply:
with a match you only read the len once